test/pytest/test_extensions_pytorch.py

```diff
@@ -0,0 +1,192 @@
+from pathlib import Path
+
+import numpy as np
+import pytest
+import torch
+
+import hls4ml
+import hls4ml.utils.torch
+
+test_root_path = Path(__file__).parent
+
+
+# PyTorch implementation of a custom layer
+class TReverse(hls4ml.utils.torch.HLS4MLModule):
+    '''PyTorch implementation of a hypothetical custom layer'''
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, inputs):
+        return torch.flip(inputs, dims=[-1])
+
+
+# hls4ml layer implementation
+class HReverse(hls4ml.model.layers.Layer):
+    '''hls4ml implementation of a hypothetical custom layer'''
+
+    def initialize(self):
+        inp = self.get_input_variable()
+        shape = inp.shape
+        dims = inp.dim_names
+        self.add_output_variable(shape, dims)
+
+
+# hls4ml optimizer to remove duplicate optimizer
+class RemoveDuplicateReverse(hls4ml.model.optimizer.OptimizerPass):
+    '''OptimizerPass to remove consecutive HReverse layers.'''
+
+    def match(self, node):
+        return isinstance(node, HReverse) and isinstance(node.get_input_node(), HReverse)
+
+    def transform(self, model, node):
+        first = node.get_input_node()
+        second = node
+
+        model.remove_node(first, rewire=True)
```

Comment:
I think the `rewire` parameter doesn't do anything anymore, might be good to just not include it here if people end up using this as an example.

Reply:
I have probably copy/pasted that line from one of the other optimizers. Should I replace all instances of this? I see there's a few around

Reply:
Might be good to clean this up in general. I'm not sure why we have this around at all anymore, since the parameter is really just a dummy according to https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/model/graph.py#L520-L537